pkg/pod/creds_init_test.go

```diff
@@ -169,3 +174,99 @@ func TestCredsInit(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckGitSSHSecret(t *testing.T) {
+	for _, tc := range []struct {
+		desc         string
+		configMap    *corev1.ConfigMap
+		secret       *corev1.Secret
+		wantError    bool
```

Comment:
I think we can simplify these test cases by just saying that, if `wantErrorMsg` is not `""`, then we expect an error, and we expect it to have that message.

Basically, `wantError` can be derived as `wantError := tc.wantErrorMsg != ""` down near line 257, and can be removed from these test cases.